src/Detector/Contracts/IDetector.cs

```diff
@@ -16,7 +16,7 @@ public interface IDetector
         /// Returns all platforms detected for the given repository context.
         /// </summary>
         /// <param name="context">The <see cref="DetectorContext"/>.</param>
-        /// <returns>A dictionary mapping the detected platforms to the version.</returns>
+        /// <returns>A list of platform detection results that contains detected platforms, versions and directories.</returns>
```

Comment:
As we keep adding richer metadata this comment might not be a place to put all that information.